reactive/kotlinx-coroutines-rx2/test/SchedulerStressTest.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.rx2
+
+import kotlinx.coroutines.*
+import org.junit.*
+import java.util.concurrent.*
+
+class SchedulerStressTest : TestBase() {
+    @Before
+    fun setup() {
+        ignoreLostThreads("RxCachedThreadScheduler-", "RxCachedWorkerPoolEvictor-", "RxSchedulerPurge-")
+    }
+
+    /**
+     * Test that we don't get an OOM if we schedule many jobs at once. It's expected that if you don't dispose that you'd
+     * see a OOM error.
+     */
+    @Test
+    fun testScheduleDirectDisposed(): Unit = runTest {
+        expect(1)
+
+        suspend fun keepMe(a: ByteArray) {
+            delay(10)
+        }
+
+        val dispatcher = currentDispatcher() as CoroutineDispatcher
+        val scheduler = dispatcher.asScheduler()
+
+        val n = 2000 * stressTestMultiplier
+        coroutineScope {
+            repeat(n) { i ->
+                val a = ByteArray(1000000) //1MB
+                val disposable = scheduler.scheduleDirect {
+                    runBlocking {
+                        keepMe(a)
+                    }
+                }
+                disposable.dispose()
+                expect(i + 2)
```

Comment:
hmm. sorry not quite following. This line is executed for n iterations.

Reply:
All the expect/finally calls in this method happen in the same coroutine. These functions are intended to use to assert execution order across multiple concurrent coroutines. 

Eg this test doesn't do anything because there is no concurrency.:
```kotlin
fun testFoo() = runTest {
  expect(1)
  for (i in 0 until 100) {
    expect(i+1)
  }
  finish(101)
}
```

Compare with this test, which asserts that the underlying dispatcher starts executing coroutines in the order in which they were launched:
```kotlin
fun testFoo() = runTest {
  expect(1)
  for (i in 0 until 100) {
    launch { expect(i+1) }
  }
  finish(101)
}
```

Reply:
Ah ok understood. Thanks for the explanation. I'll change